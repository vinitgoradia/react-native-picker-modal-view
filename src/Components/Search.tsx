// Global Imports
import * as React from 'react';
import { View, TextInput, TouchableOpacity, Image } from 'react-native';

// Local Imports
import { SearchStyle } from '@Styles';
import { ISearch } from '@Interfaces';

export class SearchComponent extends React.PureComponent<ISearch, {}> {
	public render(): JSX.Element {
		const { SearchInputProps, placeholderTextColor, onClose, setText, forceSelect, searchText, onBackRequest } = this.props;
		return (
			<View style={SearchStyle.searchArea}>
				{
					this.touchableOpacityButton(onBackRequest, require('../Assets/Images/left-arrow.png'), SearchStyle.leftBtn, SearchStyle.backButton)
				}
				<TextInput
					placeholder={searchText}
					placeholderTextColor={placeholderTextColor}
					style={[SearchStyle.textInput, forceSelect && SearchStyle.nonCloseButton]}
					underlineColorAndroid={'transparent'}
					onChangeText={(text: string) => setText(text)}
					{...SearchInputProps}
				/>
				{!forceSelect &&
					this.touchableOpacityButton(onClose, require('../Assets/Images/close.png'), SearchStyle.leftBtn, SearchStyle.closeButton)

				}
			</View>
		);
	}
	public touchableOpacityButton(onPress, imgSrc, buttonStyle, imgStyle): JSX.Element {
		return (
			<TouchableOpacity onPress={() => onPress()} style={buttonStyle}>
				<Image source={imgSrc} style={imgStyle} />
			</TouchableOpacity>
		)
	}
}
