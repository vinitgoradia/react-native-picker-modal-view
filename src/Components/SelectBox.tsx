// Global Imports
import * as React from 'react';
import { View, Text, TouchableOpacity, Image } from 'react-native';

// Local Imports
import { ISelectBoxProps } from '@Interfaces';
import { SelectBoxStyle } from '@Styles';

export class SelectBoxComponent extends React.PureComponent<ISelectBoxProps, {}> {
	public render() {
		const { openModal, selectedObject, chooseText, disabled, renderSelectView, items, isSource, isDestination } = this.props;
		const selectViewIsDisabled = (disabled || !items || items.length === 0);
		if (renderSelectView) {
			return (renderSelectView(selectViewIsDisabled, selectedObject, openModal.bind(this)))
		} else if (isSource || isDestination) {
            return (<TouchableOpacity
				activeOpacity={0.7}
				onPress={() => openModal()}
				style={[SelectBoxStyle.pressBtn, selectViewIsDisabled && SelectBoxStyle.disabledBtn, { flex: 1 }]}
			>
				<View style={[SelectBoxStyle.container, {
					flexDirection: 'row',
					justifyContent: 'flex-start',
					alignItems: 'center',
					backgroundColor: colors.clr_white,
					height: 40,
					paddingBottom: 12,
					padding: 15,
				}]}>
					<View style = {[ isSource ? styles.sourceBulletStyle : styles.destinationBulletStyle]}/>
					<Text style={[selectViewIsDisabled ? SelectBoxStyle.disabledTxt : SelectBoxStyle.chooseText]}>{
						(selectedObject && selectedObject.Name) ? selectedObject.Name : chooseText
					}</Text>
						
				</View>
			</TouchableOpacity>);
        } else {
			return (
				<TouchableOpacity
					activeOpacity={0.7}
					onPress={() => openModal()}
					style={[SelectBoxStyle.pressBtn, selectViewIsDisabled && SelectBoxStyle.disabledBtn]}
				>
					<View style={SelectBoxStyle.container}>
						<Text style={[selectViewIsDisabled ? SelectBoxStyle.disabledTxt : SelectBoxStyle.chooseText]}>{
							(selectedObject && selectedObject.Name) ? selectedObject.Name : chooseText
						}</Text>
						<Image source={require('../Assets/Images/down.png')}
							style={[SelectBoxStyle.downBtn, selectViewIsDisabled && SelectBoxStyle.disabledImage]}
						/>
					</View>
				</TouchableOpacity>
				);
			}
	}
}

const colors = {
    clr_primary: '#D90429',
    clr_white: '#FFFFFF',
};

const styles = {
    selectBoxContainerStyle: {
        flexDirection: 'row',
        justifyContent: 'flex-start',
        alignItems: 'center',
        backgroundColor: colors.clr_white,
        height: 40,
        paddingBottom: 12,
        padding: 15,
    },
    sourceBulletStyle: {
        width: 8,
        height: 8,
        backgroundColor: colors.clr_primary,
        borderWidth: 1,
        borderColor: colors.clr_primary,
        marginRight: 20,
        borderRadius: 4
    },
    destinationBulletStyle: {
        width: 8,
        height: 8,
        backgroundColor: colors.clr_white,
        borderWidth: 1,
        borderColor: colors.clr_primary,
        marginRight: 20,
        borderRadius: 4
    }
}