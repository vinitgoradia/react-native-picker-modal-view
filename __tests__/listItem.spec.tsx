import * as React from 'react';
import { shallow, ShallowWrapper, mount } from 'enzyme';
import { TouchableOpacity, View } from 'react-native';
import renderer from 'react-test-renderer';

import { ListItemComponent } from '../src/Components/ListItem';


describe('ListItemComponent', () => {
	describe('rendering', () => {
		let wrapper: ShallowWrapper;
		let mounting;
		const data = {
			Id: 1,
			Name: 'John Doe',
			Value: 'John Doe'
		}
		const spyOn = jest.fn(() => data);
		const component = (<ListItemComponent
			list={data}
			onSelectMethod={spyOn}
			defaultSelected={data}
		/>);
		beforeEach(() => {
			wrapper = shallow(component);
			mounting = mount(component);
		});

		test('should render a ListItemComponent', () => {
			const render = renderer.create(component).toJSON();
			expect(render).toMatchSnapshot();
		});

		test('should render a <TouchableOpacity />', () => {
			expect(wrapper.find(TouchableOpacity)).toHaveLength(1);
		});

		test('should render a <View />', () => {
			expect(wrapper.find(View)).toHaveLength(1);
		});

		test('should goToUp is pressed', () => {
			spyOn.mockReturnValue(data);
			wrapper.find(TouchableOpacity).first().simulate('press');
			expect(spyOn).toHaveBeenCalled();
		});

		test('should defaultSelected prop object equal to data', () => {
			expect(mounting.props().defaultSelected).toMatchObject(data);
		});

	});
});
