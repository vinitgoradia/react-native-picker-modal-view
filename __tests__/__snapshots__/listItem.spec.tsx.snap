// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemComponent rendering should render a ListItemComponent 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      John Doe
    </react-native-mock>
  </react-native-mock>
</react-native-mock>
`;
